File: dxbase/src/main/java/com/dxmovie/dxbase/binding/viewadapter/radiogroup/ViewAdapter.java
package com.dxmovie.dxbase.binding.viewadapter.radiogroup;

import androidx.annotation.IdRes;
import androidx.databinding.BindingAdapter;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import com.dxmovie.dxbase.binding.command.BindingCommand;
import com.dxmovie.dxbase.binding.command.BindingCommandWithParams;

/**
 */
public class ViewAdapter {
    @BindingAdapter(value = {"onCheckedChangedCommand"}, requireAll = false)
    public static void onCheckedChangedCommand(final RadioGroup radioGroup, final BindingCommandWithParams<String> bindingCommand) {
        radioGroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, @IdRes int checkedId) {
                RadioButton radioButton = (RadioButton) group.findViewById(checkedId);
                bindingCommand.executeAction(radioButton.getText().toString());
            }
        });
    }
}

File: dxbase/src/main/java/com/dxmovie/dxbase/binding/viewadapter/radiogroup/ViewAdapter.kt
package com.dxmovie.dxbase.binding.viewadapter.radiogroup

import android.view.View
import android.widget.RadioButton
import android.widget.RadioGroup
import androidx.databinding.BindingAdapter
import com.dxmovie.dxbase.binding.command.BindingCommandWithParams

class ViewAdapter {

    companion object {
        @JvmStatic
        @BindingAdapter(value = ["onCheckedChangedCommand"], requireAll = false)
        fun onCheckedChangedCommand(radioGroup: RadioGroup, bindingCommand: BindingCommandWithParams<String?>) {
            radioGroup.setOnCheckedChangeListener { group, checkedId ->
                val radioButton = group.findViewById<View>(checkedId) as RadioButton
                bindingCommand.executeAction(radioButton.text.toString())
            }
        }
    }
}